httpx/models.py

```diff
@@ -922,13 +923,14 @@ def raw(self):  # type: ignore
                 yield part
             await self.close()
 
-    async def next(self) -> "Response":
+    async def anext(self) -> "Response":
         """
         Get the next response from a redirect response.
         """
         if not self.is_redirect:
             raise NotRedirectResponse()
         assert self.call_next is not None
+        assert inspect.iscoroutinefunction(self.call_next)
```

Comment:
I wouldn’t bother making this functional change, or the associated `partial` change. Let’s just do a straight name change and leave it at that. We can look at any extra type smart when we introduce `.next()`

Reply:
Yeah, sure, that'd fix it *for now* — but we might bump into this again when looking at `.next()`. Anyway, will remove this check for now so we can move on.